### PR TITLE
move agents and tools out of exp

### PR DESCRIPTION
**Pull request by FluffyKebab**
Move agents and tools out of exp and fixes the lint issues in these packages.

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Describes source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.
- [x] I am an existing contributor.

### CODE REVIEW

File: agents/executor/executor.go [hunk 1]
```diff
@@ -110,3 +106,12 @@ func inputsToString(inputValues map[string]any) (map[string]string, error) {
 
 	return inputs, nil
 }
+
+func getNameToTool(t []tools.Tool) map[string]tools.Tool {
+	nameToTool := make(map[string]tools.Tool, len(t))
```
Comment (obitech) [inline agents/executor/executor.go:111]: nit:
```suggestion
	if len(t) == 0 {
		return nil
	}

	nameToTool := make(map[string]tools.Tool, len(t))
```

File: agents/executor/initialize.go [hunk 2]
```diff
@@ -58,8 +60,12 @@ func Initialize(
 	for _, opt := range opts {
 		opt(&options)
 	}
-	var agent agent.Agent
+	maxIterations, ok := options["maxIterations"].(int)
```
[the remark this answers is not in the document]
Comment (obitech) [reply agents/executor/initialize.go:63]: Hmm I'm not too familiar with the `agents` package yet. Let's get this merged then pick up the discussion in another thread 😊

File: agents/executor/executor.go (hunk 1 above)
Comment (obitech) [reply agents/executor/executor.go:111]: You're correct, it doesn't really make a difference, it's just a [style choice](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices). I marked it as nit, feel free to ignore it 😊

**Review by tmc — APPROVED** [review]
LGTM

File: agents/executor/initialize.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tmc) [reply agents/executor/initialize.go:63]: @obitech supportive.

[the remark this answers is not in the document]
Comment (tmc) [reply agents/executor/initialize.go:63]: @FluffyKebab can you sketch that idea out in a bit more detail?

**Pull request merged**